src/FormErrorInterface.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Form;
+
+/**
+ * FormErrors represents a form errors collection.
+ */
+interface FormErrorInterface
+{
+    /**
+     * Add error for the specified attribute.
+     *
+     * @param string $attribute attribute name.
+     * @param string $error attribute error message.
+     */
+    public function addError(string $attribute, string $error): void;
+
+    /**
+     * Removes errors for all attributes.
+     */
+    public function clear(): void;
+
+    /**
+     * Returns the errors for single attribute.
+     *
+     * @param string $attribute attribute name. Use null to retrieve errors for all attributes.
```

Comment:
I'd avoid using a special `null` in this case since it changes format of array returned and there introduce a separate `getAllErrors()` method instead (see below).

```suggestion
     * @param string $attribute Attribute name.
```

Reply:
done.